includes/data-events/listeners.php

```diff
@@ -100,3 +100,104 @@ function( $provider, $email, $lists_added, $lists_removed, $result ) {
 		];
 	}
 );
+
+/**
+ * For when there's a new donation through WooCommerce.
+ */
+Data_Events::register_listener(
+	'newspack_donation_order_processed',
+	'donation_new',
+	function( $order_id, $product_id ) {
+		$order = \wc_get_order( $order_id );
+		if ( ! $order ) {
+			return;
+		}
+		return [
+			'user_id'       => $order->get_customer_id(),
+			'email'         => $order->get_billing_email(),
+			'amount'        => (float) $order->get_total(),
+			'currency'      => $order->get_currency(),
+			'recurrence'    => get_post_meta( $product_id, '_subscription_period', true ),
+			'platform'      => 'newspack',
+			'platform_data' => [
+				'order_id'   => $order_id,
+				'product_id' => $product_id,
+			],
+		];
+	}
+);
+
+/**
+ * For when there's a new donation through the Stripe platform.
+ */
+Data_Events::register_listener(
+	'newspack_new_donation_woocommerce',
+	'donation_new',
+	function( $order, $client_id ) {
+		$order_id = $order->get_id();
+		return [
+			'user_id'       => $order->get_customer_id(),
+			'email'         => $order->get_billing_email(),
+			'amount'        => (float) $order->get_total(),
+			'currency'      => $order->get_currency(),
+			'recurrence'    => get_post_meta( $product_id, '_subscription_period', true ),
+			'platform'      => 'stripe',
+			'platform_data' => [
+				'order_id'   => $order_id,
+				'product_id' => \Newspack\Donations::get_order_donation_product_id( $order_id ),
+				'client_id'  => $client_id,
+			],
+		];
+	}
+);
+
+/**
+ * For when there's a new donation subscription.
+ *
+ * This will be fetched from a new donation, so we're hooking into the 'donation_new' dispatch.
+ */
+Data_Events::register_listener(
+	'newspack_data_event_dispatch_donation_new',
+	'donation_subscription_new',
+	function( $timestamp, $data ) {
+		if ( 'once' === $data['recurrence'] ) {
+			return;
+		}
+		return $data;
+	}
+);
+
+/**
+ * For when a WooCommerce Subscription is cancelled.
+ */
+Data_Events::register_listener(
+	'woocommerce_subscription_status_updated',
+	'donation_subscription_cancelled',
+	function( $subscription, $status_from, $status_to ) {
```

Comment:
Thinking from an analytics POV, it would make sense to also have the subscription details here: amount, currency, recurrence and platform.

In case we are triggering non identifiable events and analyzing them, we could then compare to see what kind of subscriptions are being cancelled.

Reply:
same for status_update.

makes sense?

Reply:
Good call! 47e503963a39cd19212588cdee5a6810cb9318e6 expands the data for both dispatches